engine/src/flutter/display_list/geometry/dl_path.cc

```diff
@@ -252,26 +268,90 @@ DlPath DlPath::operator+(const DlPath& other) const {
   return DlPath(path);
 }
 
-SkPath DlPath::ConvertToSkiaPath(const Path& path, const DlPoint& shift) {
-  SkPath sk_path;
-  sk_path.setFillType(ToSkFillType(path.GetFillType()));
+static void ReduceConic(DlPathReceiver& receiver,
+                        const DlPoint& p1,
+                        const DlPoint& cp,
+                        const DlPoint& p2,
+                        DlScalar weight) {
+  // We might eventually have conic conversion math that deals with
+  // degenerate conics gracefully (or have all receivers just handle
+  // them directly). But, until then, we will just convert them to a
+  // pair of quads and accept the results as "close enough".
+  if (p1 != cp) {
+    if (cp != p2) {
+      std::array<DlPoint, 5> points;
+      impeller::ConicPathComponent conic(p1, cp, p2, weight);
+      conic.SubdivideToQuadraticPoints(points);
+      receiver.QuadTo(points[1], points[2]);
+      receiver.QuadTo(points[3], points[4]);
+    } else {
+      receiver.LineTo(cp);
+    }
+  } else if (cp != p2) {
+    receiver.LineTo(p2);
+  }
+}
+
+class SkiaPathReceiver final : public virtual DlPathReceiver {
```

Comment:
No need for virtual inheritance here and below.

Reply:
Does that hurt anything, though?

Reply:
I tend to put that when inheriting from "interface classes" to describe the relationship better. The base class does have some helpful default implementations, but it is more intended to represent a pure interface.

Reply:
Yea, it hurts performance and it's non-idiomatic.

Reply:
virtual inheritance doesn't inline it's superclass it keeps a pointer to it.

Reply:
TIL

Reply:
We may have a lot of these throughout the DL code. :(